server/src/test/java/server/completion/CompletionProviderTest.java

```diff
@@ -0,0 +1,159 @@
+        node.addChild(main_1);
+        node.addChild(main_2);
+
+        node = tree.getRoot().addChild(lib);
+        node.addChild(lib_1);
+        node.addChild(lib_2);
+
+
+        api = new WorkspaceAPI(tree);
+        classUnderTest = Mockito.spy(CompletionProvider.getInstance());
+        tracker = Mockito.spy(DocumentTracker.getInstance());
+        Mockito.doReturn(api).when(classUnderTest).getWorkspaceAPI();
+        Mockito.doReturn(tracker).when(classUnderTest).getDocumentTracker();
+    }
+
+    @After
+    public void tearDown() {
+        classUnderTest = null;
+        api = null;
+        tree = null;
+    }
+
+    @Test
+    public void getRootFolderCompletion() throws Exception {
+//        Mockito.doReturn(
+//                "load(\"@rules_java//java:defs.bzl\", \"java_library\")\n" +
+//                        "\n" +
+//                        "package(default_visibility = [\"//visibility:public\"])\n" +
+//                        "\n" +
+//                        "java_library(\n" +
+//                        "    name = \"completion\",\n" +
+//                        "    srcs = [\n" +
+//                        "        \"CompletionProviderTest.java\",\n" +
+//                        "    ],\n" +
+//                        "    deps = [\n" +
+//                        "        \"/\",\n" +
+//                        "        \"//server/src/main/java/server/bazel/bazelWorkspaceAPI\",\n" +
+//                        "        \"//server/src/main/java/server/bazel/tree\",\n" +
+//                        "        \"//server/src/main/java/server/completion\",\n" +
+//                        "        \"//server/src/main/java/server/utils\",\n" +
+//                        "        \"//server/src/main/java/server/workspace\",\n" +
+//                        "        \"//third_party/java:gson\",\n" +
+//                        "        \"//third_party/java:guava\",\n" +
+//                        "        \"//third_party/java:jmifs\",\n" +
+//                        "        \"//third_party/java:junit\",\n" +
+//                        "        \"//third_party/java:log4j\",\n" +
+//                        "        \"//third_party/java:lsp4j\",\n" +
+//                        "        \"//third_party/java:mockito\",\n" +
+//                        "        \"//third_party/java:powermock\",\n" +
+//                        "        \"//third_party/java:powermock-junit\",\n" +
+//                        "        \"//third_party/java:powermock-mockito\",\n" +
+//                        "        \"/\"\n" +
+//                        "    ],\n" +
+//                        ")\n").when(tracker).getContents(Mockito.any());
+//        CompletionParams params = new CompletionParams(new TextDocumentIdentifier("somedocument"), new Position(26,10), new CompletionContext(CompletionTriggerKind.TriggerCharacter, "/"));
+//        CompletableFuture<Either<List<CompletionItem>, CompletionList>> future = classUnderTest.getCompletion(params);
```

Comment:
I feel like the test issues have something to do with static instances

Reply:
I've now removed all the static instances but still having a similar issue. Its weird